tests/functional/test_wps_package.py

```diff
@@ -1322,3 +1363,136 @@ def test_enum_array_and_multi_format_inputs_from_wps_xml_reference(self):
     @pytest.mark.skip(reason="not implemented")
     def test_multi_outputs_file_from_wps_xml_reference(self):
         raise NotImplementedError
+
+
+@pytest.mark.functional
+class WpsPackageAppWithS3BucketTest(WpsPackageConfigBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.settings = {
+            "weaver.wps": True,
+            "weaver.wps_output": True,
+            "weaver.wps_output_path": "/wpsoutputs",
+            "weaver.wps_output_dir": "/tmp",  # nosec: B108 # don't care hardcoded for test
+            "weaver.wps_output_s3_bucket": "wps-output-test-bucket",
+            "weaver.wps_output_s3_region": MOCK_AWS_REGION,  # must match exactly, or mock will not work
+            "weaver.wps_path": "/ows/wps",
+            "weaver.wps_restapi_path": "/",
+        }
+        super(WpsPackageAppWithS3BucketTest, cls).setUpClass()
+
+    @mocked_aws_credentials
+    @mocked_aws_s3
+    def test_execute_with_bucket(self):
```

Comment:
`test_execute_process_with_bucket` instead, to make it clearer if, for example, workflow would have particularities that would make the behaviour different

Reply:
adjusted to `test_execute_application_package_process_with_bucket` since workflows are also processes